tests/unit/test_wait.py

```diff
@@ -109,3 +109,13 @@ def test_timeout_override(run: mocks.Run, time: mocks.Time):
     assert len(run.calls) == 5
     assert time.monotonic() == 5
     assert 'mdl' in str(excinfo.value)
+
+
+def test_timeout_zero(time: mocks.Time):
```

Comment:
This gets the coverage of `_juju.py` back up to 100%.

Reply:
Lovely!

The test could be better though...
For one, the test essentially relies on `[time.monotonic(), time.monotonic()]` to produce distinct values.
That's not guaranteed in Python, even if typically true.

Reply:
No, I don't think it does. For one thing, we're using my little time mock, which only increments when you do a `time.sleep`. So both `time.monotonic()`s are guaranteed to return 0. Then the `while` condition is `time.monotonic() - start < timeout`, which is `while 0 - 0 < 0`, so the while loop never executes and it falls down to the `status is None` case.